core/src/main/java/org/mule/internal/connection/CachedConnectionProviderWrapper.java

```diff
@@ -23,13 +26,22 @@
 {
 
     private boolean disableValidation;
+    private AbstractPolicyTemplate retryPolicyTemplate;
 
-    public CachedConnectionProviderWrapper(ConnectionProvider<Config, Connection> provider, boolean disableValidation)
+    public CachedConnectionProviderWrapper(ConnectionProvider<Config, Connection> provider, boolean disableValidation, AbstractPolicyTemplate retryPolicyTemplate)
     {
         super(provider);
         this.disableValidation = disableValidation;
+        this.retryPolicyTemplate = retryPolicyTemplate;
     }
 
+    /**
+     * Delegates the responsibility of validate the connection to the delegated {@link ConnectionProvider}
```

Comment:
responsibility of validate the connection ->  responsibility of validating the connection